### PR TITLE
Profile sharing feature

### PR DESCRIPTION
**Pull request by manojVivek**
Adds an option to the profile explorer to share the selected profile.

Screenshots:
Screenshot 2022-07-07 at 8 48 31 AM
Screenshot 2022-07-07 at 8 48 37 AM
Screenshot 2022-07-07 at 3 02 57 PM

### CODE REVIEW

File: ui/packages/shared/profile/src/components/ProfileShareButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      <form className="py-2">
+        <p className="text-sm text-gray-500">
+          Note: Shared profiles can be accessed by any one with the link, even from people outside
+          your organisation.
+        </p>
+        {!isShared || error?.length > 0 ? (
+          <>
+            <p className="text-sm text-gray-500 mt-3 mb-2">Enter a description(optional)</p>
+            <textarea
+              className="border w-full"
```
Comment (yomete) [inline ui/packages/shared/profile/src/components/ProfileShareButton/index.tsx:64]:
```suggestion
              className="border w-full text-gray-500 text-sm px-2 py-2"
```

I added these style changes cos I think it makes the `textarea` field look better.

*Before* 
image


*After*
image

**Review by yomete — APPROVED** [review]
Frontend looks good to me. Just a few cosmetic changes suggested.

File: ui/packages/shared/profile/src/components/ProfileShareButton/index.tsx (hunk 1 above)
Comment (manojVivek) [reply ui/packages/shared/profile/src/components/ProfileShareButton/index.tsx]: Thanks for the suggestions, Yomi. Applied them. 

Also pushed a commit to add dark mode support for the Modal in this commit: https://github.com/parca-dev/parca/pull/1313/commits/cce0ed6edb66c7edeb18300ed1f0e7376ec17a08

I had moved the `dark` and other theme classes to `html` and `body` tags to support the `Modal` component, as it is being created as a React portal outside the `id="root"` element. 
Please let me know if you have any concerns.

**Review by monicawoj — APPROVED** [review]

**Comment by monicawoj** [comment]
Looks awesome! Love that you took the "loading" state into account for the button 👍

File: ui/packages/shared/profile/src/utils.ts [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+  metadata: RpcMetadata
+) => {
+  const req = {
+    ...request,
+    reportType: QueryRequest_ReportType.PPROF,
+  };
+
+  const {response} = await queryClient.query(req, {meta: metadata});
+  if (response.report.oneofKind !== 'pprof') {
+    console.log('Expected pprof report, got:', response.report.oneofKind);
```
[the remark this answers is not in the document]
Comment (manojVivek) [reply ui/packages/shared/profile/src/utils.ts]: Fixed it. 👍🏼

File: ui/packages/shared/profile/src/components/ProfileShareButton/index.tsx (hunk 1 above)
Comment (manojVivek) [reply ui/packages/shared/profile/src/components/ProfileShareButton/index.tsx]: Yep, I'm planning to do the same fix for the cloud app as well. 👍🏼

**Review by brancz — APPROVED** [review]
Love it!

**Pull request merged**